contracts/strategies/VariableRewardsStrategy.sol

```diff
@@ -0,0 +1,304 @@
+            "VariableRewardsStrategy::Deposit token transfer failed"
+        );
+        uint256 depositFee = _calculateDepositFee(_amount);
+        _mint(_account, getSharesForDepositTokens(_amount.sub(depositFee)));
+        _stakeDepositTokens(_amount);
+        emit Deposit(_account, _amount);
+    }
+
+    function _getDepositFeeBips() internal view virtual returns (uint256) {
+        return 0;
+    }
+
+    function _calculateDepositFee(uint256 _amount) internal view virtual returns (uint256) {
+        uint256 depositFeeBips = _getDepositFeeBips();
+        return _amount.mul(depositFeeBips).div(_bip());
+    }
+
+    function withdraw(uint256 _amount) external override {
+        uint256 depositTokenAmount = getDepositTokensForShares(_amount);
+        require(depositTokenAmount > 0, "VariableRewardsStrategy::Withdraw amount too low");
+        uint256 withdrawAmount = _withdrawFromStakingContract(_amount);
+        uint256 withdrawFee = _calculateWithdrawFee(depositTokenAmount);
+        depositToken.safeTransfer(msg.sender, withdrawAmount.sub(withdrawFee));
+        _burn(msg.sender, _amount);
+        emit Withdraw(msg.sender, depositTokenAmount);
+    }
+
+    function _getWithdrawFeeBips() internal view virtual returns (uint256) {
+        return 0;
+    }
+
+    function _calculateWithdrawFee(uint256 _amount) internal view virtual returns (uint256) {
+        uint256 withdrawFeeBips = _getWithdrawFeeBips();
+        return _amount.mul(withdrawFeeBips).div(_bip());
+    }
+
+    function reinvest() external override onlyEOA {
+        _reinvest();
+    }
+
+    function _convertRewardsIntoWAVAX() private returns (uint256) {
+        uint256 avaxAmount = WAVAX.balanceOf(address(this));
+        uint256 count = rewardCount;
+        for (uint256 i = 0; i < count; i++) {
+            address reward = supportedRewards[i];
+            if (reward == address(WAVAX)) {
+                uint256 balance = address(this).balance;
+                if (balance > 0) {
+                    WAVAX.deposit{value: balance}();
+                    avaxAmount = avaxAmount.add(balance);
+                }
+                continue;
+            }
+            uint256 amount = IERC20(reward).balanceOf(address(this));
+            if (amount > 0) {
+                address swapPair = rewardSwapPairs[reward];
+                if (swapPair > address(0)) {
+                    avaxAmount = avaxAmount.add(DexLibrary.swap(amount, reward, address(rewardToken), IPair(swapPair)));
+                }
+            }
+        }
+        return avaxAmount;
+    }
+
+    /**
+     * @notice Reinvest rewards from staking contract to deposit tokens
+     * @dev Reverts if the expected amount of tokens are not returned from the staking contract
+     */
+    function _reinvest() private {
+        _getRewards();
+        uint256 amount = _convertRewardsIntoWAVAX();
+        require(amount >= MIN_TOKENS_TO_REINVEST, "VariableRewardsStrategy::Reinvest amount too low");
```

Comment:
if `MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST` is set and `_reinvest()` is called during deposit, this could cause deposits to fail

Reply:
fixed